### PR TITLE
refactored compute forward to not pass in the src tensors each time

### PR DESCRIPTION
**Pull request by siddharthvader**
intended to address issue #724

### CODE REVIEW

File: src/ggml.c [hunk 1]
```diff
@@ -2607,7 +2607,7 @@ static struct ggml_tensor * ggml_new_tensor_impl(
         /*.nb           =*/ { 0, 0, 0, 0 },
         /*.op           =*/ GGML_OP_NONE,
         /*.op_params    =*/ { 0 },
-        /*.flags        =*/ 0,
+        /*.is_param     =*/ false,
```
Comment (slaren) [inline src/ggml.c:2610]: It looks like something went wrong while merging with the latest changes, this and all other uses of `flags` shouldn't be changed.

Comment (siddharthvader) [reply src/ggml.c:2610]: hmm got it, will take a look

Comment (siddharthvader) [reply src/ggml.c]: okay I should have fixed all the instances where there were merge issues affecting `flags`. the code builds now after running `make -j4 gpt-2-backend gpt-j` so I think it should be in good shape now!

**Comment by siddharthvader** [comment]
looking into why the customop test is failing

**Review by slaren — APPROVED** [review]

**Pull request merged**